Samples/MapScenarios/MapScenarios/Scenarios/WalkTimeViewController.swift

```diff
@@ -23,51 +23,139 @@ class WalkTimeViewController: UIViewController {
     let locationManager = CLLocationManager()
     var firstLocationAcquired = false
     
+    // MARK: Walk Time
+    // Walk speed is in meters/second
+    var walkSpeed: Double {
+        if let calculatedWalkSpeed = calculatedWalkSpeed, acceptedWalkSpeedRange.contains(calculatedWalkSpeed) {
+            return calculatedWalkSpeed
+        }
+        return defaultWalkSpeed
+    }
+    let defaultWalkSpeed = 1.4
+    var calculatedWalkSpeed: Double?
+    let acceptedWalkSpeedRange = 0.5..<5.0
+    var estimatedWalkTime: Double?
+    var lastLocationUpdates = [CLLocation]()
+    let walkTimeLabel = UILabel()
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         
         navigationItem.title = "Walk Time Calculation"
         
-        if applicationId.count > 0 && accessKey.count > 0 && signatureKey.count > 0 && buildingIdentifier != 0 {
-            PWCore.setApplicationID(applicationId, accessKey: accessKey, signatureKey: signatureKey)
-        } else {
-            fatalError("applicationId, accessKey, signatureKey, and buildingIdentifier must be set")
+        if !validateBuildingSetting(appId: applicationId, accessKey: accessKey, signatureKey: signatureKey, buildingId: buildingIdentifier) {
```

Comment:
I see this code a lot. Can we refactor into a base view controller that we just reuse?

Reply:
The reason for a lot of the duplicated code was that the original intent of these scenarios was that you could copy-paste the entire view controller and it would work as-is without having to look around in other files to see what you missed. Now that we're adding more complex, shared UI we're breaking that paradigm so let's discuss the best way to handle it (base view controller, protocols, etc)